packages/venia-concept/src/actions/catalog/asyncActions.js

```diff
@@ -1,4 +1,85 @@
 import actions from './actions';
+import mockData from './mockData';
+
+export const serialize = (params, keys = [], isArray = false) => {
+    const p = Object.keys(params)
+        .map(key => {
+            let val = params[key];
+            if (
+                '[object Object]' === Object.prototype.toString.call(val) ||
+                Array.isArray(val)
+            ) {
+                if (val.length === 0) return null;
+
+                if (Array.isArray(params)) {
+                    keys.push('');
+                } else {
+                    keys.push(key);
+                }
+                return serialize(val, keys, Array.isArray(val));
+            } else {
+                let tKey = key;
+
+                if (keys.length > 0) {
+                    const tKeys = isArray
+                        ? keys.filter(v => v != '')
+                        : [...keys, key].filter(v => v != '');
+                    tKey = tKeys.reduce((str, k) => {
+                        return '' === str ? k : `${str}[${k}]`;
+                    }, '');
+                }
+                if (isArray) {
+                    return `${tKey}[]=${val}`;
+                } else {
+                    return `${tKey}=${val}`;
+                }
+            }
+        })
+        .filter(Boolean)
+        .join('&');
+
+    keys.pop();
+    return p;
+};
+
+const updateCatalogUrl = (filters, history) =>
+    history.push('?' + serialize(filters));
+
+export const filterAdd = ({ group, title, value }, history) =>
+    async function thunk(dispatch, getState) {
+        const {
+            catalog: { chosenFilterOptions }
+        } = getState();
+
+        const oldState = chosenFilterOptions[group] || [];
+        const newState = oldState.concat({ title, value });
+
+        dispatch(actions.filterOption.add({ newState, group }));
```

Comment:
The `filterOption.add` and `filterOption.remove` functions are identical. Would it be OK to simplify into a single `filterOption.update` function that both this `filterAdd` and the below `filterRemove` function use?

Reply:
Resolved in latest update.